fontc/src/cli_args.rs

```diff
@@ -227,12 +186,36 @@ impl Serialize for ValidatedRegex {
         self.0.as_str().serialize(serializer)
     }
 }
+
+impl TryInto<fontc::Args> for Args {
+    type Error = Error;
+
+    fn try_into(self) -> Result<fontc::Args, Self::Error> {
+        let flags = self.flags();
+        let input = self.source()?;
+        let timing_file = self.emit_timing.then(|| self.build_dir.join("threads.svg"));
+        let debug_dir = self.emit_debug.then(|| self.build_dir.clone());
+        let ir_dir = self.emit_ir.then(|| self.build_dir.clone());
+        Ok(fontc::Args {
+            flags,
+            skip_features: self.skip_features,
+            output_file: self
+                .output_file
+                .or_else(|| Some(self.build_dir.join("font.ttf"))),
```

Comment:
there's the same problem I identified in Simon's PR when --emit-ir is used in combination with a custom --output-file.
When IR is enabled, ir_dir is set to build_dir; the font gets persisted (via BePersistentStorage) to build_dir/font.ttf regardless of output_file. But `write_font_file` function checks if `!output_file.exists()` and fails if user passed --output-file to something that != build_dir/font.ttf.

Maybe we should simply copy/rename font from ir_dir/font.ttf to wherever output_file is when ir_dir is Some.